src/encoding_binding.cc

```diff
@@ -0,0 +1,211 @@
+#include "encoding_binding.h"
+#include "env-inl.h"
+#include "node_errors.h"
+#include "node_external_reference.h"
+#include "simdutf.h"
+#include "string_bytes.h"
+#include "v8.h"
+
+#include <cstdint>
+
+namespace node {
+namespace encoding_binding {
+
+using v8::ArrayBuffer;
+using v8::BackingStore;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Isolate;
+using v8::Local;
+using v8::MaybeLocal;
+using v8::Object;
+using v8::String;
+using v8::Uint8Array;
+using v8::Value;
+
+void BindingData::MemoryInfo(MemoryTracker* tracker) const {
+  tracker->TrackField("encode_into_results_buffer",
+                      encode_into_results_buffer_);
+}
+
+BindingData::BindingData(Realm* realm, v8::Local<v8::Object> object)
+    : SnapshotableObject(realm, object, type_int),
+      encode_into_results_buffer_(realm->isolate(), kEncodeIntoResultsLength) {
+  object
+      ->Set(realm->context(),
+            FIXED_ONE_BYTE_STRING(realm->isolate(), "encodeIntoResults"),
+            encode_into_results_buffer_.GetJSArray())
+      .Check();
+}
+
+bool BindingData::PrepareForSerialization(Local<Context> context,
+                                          v8::SnapshotCreator* creator) {
+  // We'll just re-initialize the buffers in the constructor since their
+  // contents can be thrown away once consumed in the previous call.
+  encode_into_results_buffer_.Release();
+  // Return true because we need to maintain the reference to the binding from
+  // JS land.
+  return true;
+}
+
+InternalFieldInfoBase* BindingData::Serialize(int index) {
+  DCHECK_EQ(index, BaseObject::kEmbedderType);
+  InternalFieldInfo* info =
+      InternalFieldInfoBase::New<InternalFieldInfo>(type());
+  return info;
+}
+
+void BindingData::Deserialize(Local<Context> context,
+                              Local<Object> holder,
+                              int index,
+                              InternalFieldInfoBase* info) {
+  DCHECK_EQ(index, BaseObject::kEmbedderType);
+  v8::HandleScope scope(context->GetIsolate());
+  Realm* realm = Realm::GetCurrent(context);
+  // Recreate the buffer in the constructor.
+  BindingData* binding = realm->AddBindingData<BindingData>(context, holder);
+  CHECK_NOT_NULL(binding);
+}
+
+void BindingData::EncodeInto(const FunctionCallbackInfo<Value>& args) {
```

Comment:
Out of curiosity I exploring places where fast API could be applied, came across this function and PR, I experimented a bit it seems ```v8::FastApiTypedArray<uint8_t>&``` is indeed supported and it does get caught in fast call, but I am not sure how one could extract ```->Buffer()``` from ```v8::FastApiTypedArray<uint8_t>&```, is that even possible? just wondering do let me know your thoughts cc @anonrig @joyeecheung 

Thank You!